### PR TITLE
Feat:builtin signal

### LINKED ISSUE #11: Feature: Handle ctrl-C, ctrl-D and ctrl-\
Handle ctrl-C, ctrl-D and ctrl-\ which should behave like in bash.

In interactive mode:
◦ ctrl-C displays a new prompt on a new line.
◦ ctrl-D exits the shell.
◦ ctrl-\ does nothing

[Signals in C](https://www.geeksforgeeks.org/signals-c-language/)

[Sending and Handling Signals in C (kill, signal, sigaction)](https://www.youtube.com/watch?v=83M5-NPDeWs&ab_channel=JacobSorber)

### PR DESCRIPTION
**Pull request by iaurg**
- [x] ctrl-C displays a new prompt on a new line.
- [x] ctrl-D exits the shell.
- [x] ctrl-\ does nothing
Closes #11

### CODE REVIEW

File: src/prompt.c [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+{
+	char	cwd[1025];
+	char	*tmp;
+	char	*other;
+	char	*path;
+	char	*tmp_user;
+
+	getcwd(cwd, 1024);
+	path = get_last_slash_arg(cwd);
+	tmp_user = ft_strdup(getenv("USER"));
```
Comment (iaurg) [inline src/prompt.c:25]: Signal handlers don't accept params, how can I get here with our envp?

File: src/execute_system.c [hunk 2]
```diff
@@ -77,6 +77,7 @@ int	system_exec(t_command *cmd)
 	int		status;
 
 	pid = fork();
+	signal(SIGINT, child_signal_handler);
```
Comment (iaurg) [inline src/execute_system.c:109]: What's the better way to test this signal? I've tested in many ways and all worked.

Comment (vcwild) [reply src/execute_system.c:109]: I don't know. What is this supposed to do?

Comment (iaurg) [reply src/execute_system.c:109]: Handle signals coming from child proccess

Comment (vcwild) [reply src/execute_system.c:109]: Maybe we should test it separate as a main proccess and see how it handles signals

**Review by vcwild — APPROVED** [review]
LGTM!

**Pull request merged**